src/lib/y2network/proposal_settings.rb

```diff
@@ -164,6 +190,34 @@ def create_instance
 
   private
 
+    # Convenience method to load some features from a installation control file section
+    #
+    # @param source [Hash] network section to be loaded
+    def load_features(source = network_section)
+      return unless source.is_a?(Hash)
+
+      source.keys.each { |k| load_feature(k, k, source: source) if respond_to?("#{k}=") }
+    end
+
+    # Reads a feature from a given hash and assign it to the corresponding object attribute
+    #
+    # @param feature [String, Symbol] feature name to be read
+    # @param to [String, Symbol] attribute name where to store the feature value
+    # @param source [Hash] from where to read the feature
+    def load_feature(feature, to, source: network_section)
+      value = source[feature.to_s]
+      public_send("#{to}=", value) unless value.nil?
+    end
```

Comment:
NP: What about moving the condition to the #load_feature method?

```suggestion
      source.keys.each { |k| load_feature(k, k, source: source) }
    end

    # Reads a feature from a given hash and assign it to the corresponding object attribute
    #
    # @param feature [String, Symbol] feature name to be read
    # @param to [String, Symbol] attribute name where to store the feature value
    # @param source [Hash] from where to read the feature
    def load_feature(feature, to, source: network_section)
      return unless respond_to?("#{to}=")
      
      value = source[feature.to_s]
      public_send("#{to}=", value) unless value.nil?
    end
```

Reply:
And reading the code maybe makes sense to set the value always. I mean, in case you change the role and it has different values where some of the attributes is not set, then we should not modify it.